tools/release_generator/release.py

```diff
@@ -225,18 +226,26 @@ def makeSubDirectoriesFromList(directoryList, upperDirectory, HigherLevel):
         makeReleaseNoteSubfolders(combinedDirectory, HigherLevel)
 
 
+def subfolder_creation(directory, HigherLevel, folder):
+    subfolderName = HigherLevel / directory / folder
+    subfolderName.mkdir(parents=True, exist_ok=True)
+    makeGitkeep(subfolderName)
+
+
 def makeReleaseNoteSubfolders(directory, HigherLevel):
```

Comment:
Happy to merge this in as it is if preferred, as it sounds like this is doing what we need it to. However I just thought I'd mention some thoughts on whether we might be able to simplify and streamline this method a bit to make it easier to understand and maintain. I may not be following the logic correctly, so apologies if that's the case, but would the following achieve the same result?

```
def makeReleaseNoteSubfolders(directory, HigherLevel):
    directoryStr = str(directory)
    
    if "Muon" in directoryStr:
	    folders_to_create = muon_subfolders
    elif "Framework" in directoryStr and ("Algorithm" in directoryStr or "Fit_Functions" in directoryStr):
        folders_to_create = deprecated_subfolders
    else:
        folders_to_create = subfolders
    
    for folder in folders_to_create:
	    subfolder_creation(directory, HigherLevel, folder)
```

Again I may be following the code in the PR incorrectly, but it looks to me like the nested looping would mean that we're trying to duplicate the folder creation for Muon and deprecated subfolders, which doesn't actually result in duplicated subfolders but would be unnecessary extra effort.

Reply:
If the above was a possible alternative, you could even consider pulling the subfolder finding into it's own method so you would have:

```
def getSubfoldersForDirectory(director):
    if "Muon" in directoryStr:
	    return muon_subfolders
    
    if "Framework" in directoryStr and ("Algorithm" in directoryStr or "Fit_Functions" in directoryStr):
	    return deprecated_subfolders

    return subfolders

def makeReleaseNoteSubfolders(directory, HigherLevel):    
    for folder in getSubfoldersForDirectory(directory):
	    subfolder_creation(directory, HigherLevel, folder)
```

Reply:
Thank you @rbauststfc . The suggestions do made the code cleaner and require less effort. Having tested them I can confirm they do what is needed to make the right folders in the right place :) I've made the changes and will push a new commit